### ISSUE TITLE
view for creating half-open Integral ranges

### ISSUE THREAD

**Issue by gnzlbg** [ericniebler/range-v3#122]
There are currently two views for creating closed (aka inclusive) `Integral` ranges (`view::iota` and `view::ints`). There aren't any utilities for creating half-open `Integral` ranges. Most range interfaces offer an utility for half-open `Integral` ranges: e.g. `boost::counting_range`, Python's `xrange`/`range`, D's `iota`, ... I'd argue that half-open integer ranges are more useful than inclusive ones (e.g. indexing over a container is terribly common), but inclusive ones are also very useful (e.g. Matlab's `linspace` is inclusive and issue #102 ).

**Naming**: no idea.  In #16  I used `irng` to name an utility function for creating a half-open integer range but I don't like this name. I wish that both `ints` and `iota` would return half-open ranges, and that there would be a `linspace`-like function for creating inclusive ranges as mentioned in issue #102. 

**Notes**: for indexing a container `irng(0, size)` where `0 : int` and `size : std::size_t` is very common so the view should handle this.

I use this _a lot_ via: 

```c++
template <class From, class To, class CT = std::common_type_t<From, To>>
constexpr auto irng(From from, To to) {
  return take(iota(CT(from)), CT(to) - CT(from));
}
```

but this is not the best solution to the problem.

**Comment by ericniebler** [comment]
Agree there's a need for both. Not sure we need a differently named view for it or whether the existing ones can be pressed into service. Maybe `view::ints(a, b)` is open and `view::ints({a,b})` is closed. Or `view::ints(closed{a,b})`.

**Comment by gnzlbg** [comment]
I don't know yet about the `view::ints({a,b})` syntax (might be more trouble than it's worth), but I like that you also prefer half-open ranges from `ints` by default.

**Comment by ericniebler** [comment]
... or `view::ints[{a,b}]` for closed. Square brackets are used to denote a closed range. Too bad I can't use `view::ints[a,b)` for half-open. ;-)

**Comment by viboes** [comment]
... or `view::ints(closed{a}, open{b})`. The question is if a default for `view::ints(a, b)` would be needed?

**Comment by gnzlbg** [comment]
_Iff_ a default is chosen, I think that it should be half-open. There are multiple good reasons for this (ability to handle empty ranges, indexing a container, ...) but the most important I think is consistency with all other ranges which are half-open.

**Comment by ericniebler** [comment]
Fixed in 963ac26
